src/BuildScriptGeneratorCli/Commands/DetectCommand.cs

```diff
@@ -0,0 +1,123 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator.Common;
+using Microsoft.Oryx.Detector;
+using Newtonsoft.Json;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command(Name, Description = "Detect all platforms and versions in the given app source directory.")]
+    internal class DetectCommand : CommandBase
+    {
+        public const string Name = "detect";
+
+        [Argument(0, Description = "The source directory. If no value is provided, the current directory is used.")]
+        [DirectoryExists]
+        public string SourceDir { get; set; }
+
+        [Option("--json", Description = "Output the detected platform data in JSON format.")]
+        public bool OutputJson { get; set; }
+
+        internal override int Execute(IServiceProvider serviceProvider, IConsole console)
+        {
+            var loggerFactory = serviceProvider.GetRequiredService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger<DetectCommand>();
+            var sourceRepo = new LocalSourceRepo(SourceDir, loggerFactory);
+            var ctx = new DetectorContext
+            {
+                SourceRepo = sourceRepo,
+            };
+
+            var detector = serviceProvider.GetRequiredService<IDetector>();
+            var detectedPlatforms = detector.GetAllDetectedPlatforms(ctx);
+            using (var timedEvent = logger.LogTimedEvent("DetectCommand"))
+            {
+                if (detectedPlatforms != null && detectedPlatforms.Any())
+                {
+                    if (OutputJson)
```

Comment:
if nothing is detected, we could return an empty JSON array or object to facilitate the parsing by the caller